cpp/ql/src/Security/CWE/CWE-191/UnsignedDifferenceExpressionComparedZero.ql

```diff
@@ -36,47 +38,56 @@ predicate isGuarded(SubExpr sub, Expr left, Expr right) {
  * `sub.getLeftOperand()`.
  */
 predicate exprIsSubLeftOrLess(SubExpr sub, DataFlow::Node n) {
-  n.asExpr() = sub.getLeftOperand()
-  or
-  exists(DataFlow::Node other |
-    // dataflow
-    exprIsSubLeftOrLess(sub, other) and
-    (
-      DataFlow::localFlowStep(n, other) or
-      DataFlow::localFlowStep(other, n)
+  interestingSubExpr(sub, _) and // Manual magic
```

Comment:
Because `exprIsSubLeftOrLess` is recursive, this is not simply a copy of the call to `interestingSubExpr` in the query `where` block - it also enforces constraints on each recursive step that weren't there before.  I'd like to see either reasoning or a diff query demonstrating that this hasn't changed behaviour in a way we care about.

Reply:
The `sub` column never changes in any of the recursive calls, so the constraint placed on `sub` here shouldn't make any difference since the same `SubExpr` will be restricted in the `where` block anyway (unless I'm missing something in the query?)

Reply:
Ah I see, the `_` is important here.  :+1: